src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../PeopleTable';
+import { PeopleContext } from '../../peopleContext';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setError(false);
+    setLoading(true);
+
+    getPeople()
+      .then(res => setPeople(res))
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="box table-container">
+          {loading && <Loader />}
+
+          {error && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {!loading && !error && people.length <= 0 && (
+            <p data-cy="noPeopleMessage">There are no people on the server</p>
+          )}
+          <PeopleContext.Provider value={people}>
+            {<PeopleTable />}
+          </PeopleContext.Provider>
```

Comment:
It's better to wrap the entire component in the PeopleContext.Provider to avoid unnecessary re-renders. This way, the context value will only change when the 'people' state changes.